packages/blockchain/src/consensus/clique.ts

```diff
@@ -455,6 +478,7 @@ export class CliqueConsensus implements Consensus {
     this._cliqueLatestSignerStates = this._cliqueLatestSignerStates.filter(
       (s) => s[0] <= blockNumber
     )
+    this._cliqueLatestSignerStates.sort((a, b) => (a[0] > b[0] ? 1 : -1))
```

Comment:
On each reorg they will be out-of-order and will mess up the latest states (this fixes reorgs in case voting changes). However since this is in the delete method and the assumption is that the signer state is always sorted, this should only pop items from the end of the array, so might indeed remove.